include/hlslib/xilinx/OpenCL.h

```diff
@@ -28,10 +28,50 @@ namespace ocl {
 
 #if HLSLIB_LEGACY_SDX == 0
 constexpr auto kXilinxMemPointer = CL_MEM_EXT_PTR_XILINX;
-constexpr auto kMemoryBank0 = XCL_MEM_DDR_BANK0;
-constexpr auto kMemoryBank1 = XCL_MEM_DDR_BANK1;
-constexpr auto kMemoryBank2 = XCL_MEM_DDR_BANK2;
-constexpr auto kMemoryBank3 = XCL_MEM_DDR_BANK3;
+
+/*
+The xilinx alveo u280 fpga has different flags for the DRAM banks, then the existing 
+fpga's. Therefore for Xilinx the DDR-flags are now initialized and stored as part of the
+Contex-class.
+*/
+class DDRBankFlags {
+public:
+  DDRBankFlags(const std::string &device_name) {
+    if (device_name.find("xilinx_u280") != std::string::npos) {
+      memory_bank_0 = XCL_MEM_TOPOLOGY | 32;
+      memory_bank_1 = XCL_MEM_TOPOLOGY | 33;
+      memory_bank_2 = -1;
+      memory_bank_3 = -1;
+    } else {
+      DefaultAssignment();
+    }
+  }
+
+  DDRBankFlags() { DefaultAssignment(); }
+
+  inline int kMemoryBank0() const { return memory_bank_0; }
```

Comment:
Member variables are `memory_bank_0_`, accessor function can then be `memory_bank_0() const { return memory_bank_0_; }`